pkg/build/builder/cmd/scmauth/util.go

```diff
@@ -3,23 +3,53 @@ package scmauth
 import (
 	"fmt"
 	"io/ioutil"
+	"net/url"
 	"os"
 	"path/filepath"
 	"strings"
 
-	builder "github.com/openshift/builder/pkg/build/builder"
+	buildv1 "github.com/openshift/api/build/v1"
+	"github.com/openshift/builder/pkg/build/builder"
 	utillog "github.com/openshift/builder/pkg/build/builder/util/log"
 )
 
 var log = utillog.ToFile(os.Stderr, 2)
 
-func createGitConfig(includePath string, context SCMAuthContext) error {
+func createGitConfig(includePath string, context SCMAuthContext, gitSource *buildv1.GitBuildSource) error {
 	tempDir, err := ioutil.TempDir("", "git")
 	if err != nil {
 		return err
 	}
 	gitconfig := filepath.Join(tempDir, ".gitconfig")
-	content := fmt.Sprintf("[include]\npath = %s\n", includePath)
+	content := ""
+	if len(includePath) > 0 {
+		content = fmt.Sprintf("[include]\npath = %s\n", includePath)
+	}
+	if gitSource != nil {
+		if gitSource.HTTPProxy != nil && len(*gitSource.HTTPProxy) > 0 {
+			content = content + fmt.Sprintf("[http]\nproxy = %s\n", *gitSource.HTTPProxy)
+		}
+		if gitSource.HTTPSProxy != nil && len(*gitSource.HTTPSProxy) > 0 {
+			content = content + fmt.Sprintf("[https]\nproxy = %s\n", *gitSource.HTTPSProxy)
+		}
+		if gitSource.NoProxy != nil && len(*gitSource.NoProxy) > 0 {
+			gitURL, err := url.Parse(gitSource.URI)
+			if err != nil {
+				return err
+			}
+			noProxies := strings.Split(*gitSource.NoProxy, ",")
+			for _, noProxy := range noProxies {
+				if strings.HasSuffix(gitURL.Host, noProxy) {
```

Comment:
I may have led you astray here. Will this error out for an SCP-style URI? ex: `git@github.com/...`

Reply:
Yeah it does not error out, but it does not do what we want.  The host field is blank.  From goland playground:

```
package main

import (
	"fmt"
	"log"
	"net/url"
)

func main() {
	u, err := url.Parse("git@github.com/openshift/origin")
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println("string")
	fmt.Println(u.String())
	fmt.Println("just host")
	fmt.Println(u.Host)
}

```
string
git@github.com/openshift/origin
just host


```

```

Reply:
per conversation in slack ... I'll push up a commit soon that checks the scheme to see if http or https so as to bypass git@ or git:  (which both are ssh / key related).

going to wait until the current e2e-*-builds runs are complete first though ... just curious

Reply:
ok all the proxy related tests in https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_builder/187/pull-ci-openshift-builder-master-e2e-aws-builds/1328407467607986176 passed 

that said, I'll push up the new form in a moment @adambkaplan